lib/Cake/Network/Email/CakeEmail.php

```diff
@@ -1717,6 +1721,27 @@ protected function _renderTemplates($content) {
 			$rendered[$type] = implode("\n", $rendered[$type]);
 			$rendered[$type] = rtrim($rendered[$type], "\n");
 		}
+
+		/* Embed images inline in html templates */
+		if (!empty($rendered['html'])) {
```

Comment:
Of course I understand the security issues. Clearly a high memory consumption can cause security problems, especially in a function that is not intended for serialization user data.

Thanks to having done this pull I could improve my code in a very short time to find mistakes he had overlooked.

In my case the email template loads an element with the signature. This signature is one that contains an image I want to embed.

At first created the signature-element with the image as "<img src =" cid: uid ">" where uid was a unique-id precalculated. Then attached the image with the same uid as name ... evidentement worked, but the usability was ZERO.

The "Cake / View" can reliably detect images to embed, as it controls the "safe" part of the template and the "unsafe" user. However you do not have access to the "addAttachments" method of the "Cake \ Mailer \ Email".

That's why I ended up doing the "Cake \ Mailer \ Email" do the job. Since I consider it who has to perform this task.